src/TorchSharp/Utils/tensorboard/Summary.cs

```diff
@@ -0,0 +1,198 @@
+                        tensor = tensor.to_type(ScalarType.Float32);
+                        tensor = (tensor * scale_factor).clip(0, 255).to_type(ScalarType.Byte);
+                        Tensorboard.Summary.Types.Image image = make_image(tensor, rescale);
+                        var summary = new Tensorboard.Summary();
+                        summary.Value.Add(new Tensorboard.Summary.Types.Value() { Tag = tag, Image = image });
+                        return summary;
+                    }
+
+                    /// <summary>
+                    /// Outputs a `Summary` protocol buffer with images.
+                    /// </summary>
+                    /// <param name="tag"> A name for the generated node. Will also serve as a series name in TensorBoard. </param>
+                    /// <param name="file_name"> local image filename </param>
+                    /// <param name="rescale"> Rescale image size </param>
+                    /// <returns> A scalar `Tensor` of type `string`. The serialized `Summary` protocol buffer. </returns>
+                    public static Tensorboard.Summary image(string tag, string file_name, double rescale = 1)
+                    {
+                        using var img = SKBitmap.Decode(file_name);
+                        Tensorboard.Summary.Types.Image image = make_image(img, rescale);
+                        var summary = new Tensorboard.Summary();
+                        summary.Value.Add(new Tensorboard.Summary.Types.Value() { Tag = tag, Image = image });
+                        return summary;
+                    }
+
+                    /// <summary>
+                    /// Convert a tensor representation of an image to Image protobuf
+                    /// 
+                    /// https://github.com/pytorch/pytorch/blob/master/torch/utils/tensorboard/summary.py#L481
+                    /// </summary>
+                    /// <param name="tensor"> HWC(0~255) image tensor </param>
+                    /// <param name="rescale"> Rescale image size </param>
+                    /// <returns></returns>
+                    public static Tensorboard.Summary.Types.Image make_image(Tensor tensor, double rescale = 1)
+                    {
+                        using SKBitmap skBmp = TensorToSKBitmap(tensor);
+                        return make_image(skBmp, rescale);
+                    }
+
+                    /// <summary>
+                    /// Convert an image to Image protobuf
+                    /// 
+                    /// https://github.com/pytorch/pytorch/blob/master/torch/utils/tensorboard/summary.py#L495
+                    /// </summary>
+                    /// <param name="img"> Image </param>
+                    /// <param name="rescale"> Rescale image size </param>
+                    /// <returns></returns>
+                    public static Tensorboard.Summary.Types.Image make_image(SKBitmap img, double rescale = 1)
```

Comment:
Ideally, we would avoid exposing SkiaSharp in public APIs.

Reply:
I made this method internal.